src/app/common/components/mat-controls/transaction-type-mat-select/transaction-type-mat-select.component.ts

```diff
@@ -0,0 +1,39 @@
+import { Component, EventEmitter, Input, OnDestroy, OnInit, Output } from '@angular/core';
+import { FormControl } from '@angular/forms';
+import { Subscription } from 'rxjs';
+import { WalletTransactionType } from 'src/app/domains/wallets/domains.wallets.types';
+
+export type WalletSelectionValue = WalletTransactionType | '';
+
+@Component({
+  selector: 'app-transaction-type-mat-select',
+  templateUrl: './transaction-type-mat-select.component.html',
+  styleUrls: [ './transaction-type-mat-select.component.scss' ],
+})
+export class TransactionTypeMatSelectComponent implements OnInit, OnDestroy {
+  @Output() public setTransactionType = new EventEmitter<WalletSelectionValue>();
```

Comment:
Already added comment about this but seems like this component is making your life more difficult than simpler :P
For example check available transaction types, why we need `All transactions` for transaction editor?